site/docs/relations/logical_relations.md

```diff
@@ -268,14 +268,23 @@ The set operation encompasses several set-level operations that support combinin
+|----------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------| -----------------------------
+| Minus (Primary)            | Returns all records from the primary input excluding any matching rows from secondary inputs, removing duplicate rows within or across sets.<br/>Each value is treated as a unique member of the set, so duplicates in the first set don’t affect the result. | MINUS<br/>&nbsp;&nbsp;m:  {1, 2, 2, 3, 3, 3, 4}<br/>&nbsp;&nbsp;n1: {1, 2}<br/>&nbsp;&nbsp;n2: {3}<br/>YIELDS<br/>{4}                                            | The same as the primary input.                                                                                                                                                                     
+| Minus (Primary All)        | Returns all records from the primary input excluding any matching records from secondary inputs.<br/>For each specific record returned, the output contains max(0, m - sum(n1, n2, …, n)) copies.                                                             | MINUS ALL<br/>&nbsp;&nbsp;m:  {1, 2, 2, 3, 3, 3, 3}<br/>&nbsp;&nbsp;n1: {1, 2, 3, 4}<br/>&nbsp;&nbsp;n2: {3}<br/>YIELDS<br/>{2, 3, 3}                            | The same as the primary input.
+| Minus (Multiset)           | Returns all records from the primary input excluding any records that are included in *all* secondary inputs.                                                                                                                                                 | MINUS MULTISET<br/>&nbsp;&nbsp;m:  {1, 2, 3, 4}<br/>&nbsp;&nbsp;n1: {1, 2}<br/>&nbsp;&nbsp;n2: {1, 2, 3}<br/>YIELDS<br/>{4}                                      | The same as the primary input.
+| Intersection (Primary)     | Returns all records from the primary input that are present in every secondary input, removing duplicate rows within or across sets.                                                                                                                          | INTERSECT<br/>&nbsp;&nbsp;m:  {1, 2, 2, 3, 3, 3, 4}<br/>&nbsp;&nbsp;n1: {1, 2, 3, 5}<br/>&nbsp;&nbsp;n2: {2, 3, 6}<br/>YIELDS<br/>{2, 3}                         | If a field is nullable in the primary input and in any of the secondary inputs, it is nullable in the output. 
```

Comment:
> Returns all records from the primary input that are present in _**every**_ secondary input.

This used to be **any** secondary input and not **every** secondary input, so this changes the definition beyond just making deduplication explicit.

 I'm also realising that while
```sql
<A> INTERSECT <B>
```
can map to `SET_OP_INTERSECTION_PRIMARY`

```
<A> INTERSECT <B> INTERSECT <C>`
```
doesn't, because the behaviour of chained intersects like this is consistent with `SET_OP_INTERSECTION_MULTISET`. SQL `INTERSECT` should generally map to this.

Taking this further,
```sql
<A> INTERSECT ALL <B> INTERSECT ALL <C>
```
would need to be mapped to something like
```
SET_OP_INTERSECTION_MULTISET_ALL
```
instead of 
```
SET_OP_INTERSECTION_PRIMARY_ALL
```

I think it might make sense to define
```
SET_OP_INTERSECTION_MULTISET_ALL
```
and not
```
SET_OP_INTERSECTION_PRIMARY_ALL
```

Reply:
I agree that the new definition of `Intersection (Primary)` is incorrect.

Why aren't we adding both ` SET_OP_INTERSECTION_MULTISET_ALL` and `SET_OP_INTERSECTION_PRIMARY_ALL`

I agree with your logic that, if we are adding only one, `MULTISET_ALL` makes more sense.

Reply:
FWIW: Ah, I think I see now, the duplicate handling behavior of `SET_OP_INTERSECTION_PRIMARY_ALL` is ambiguous.  I could interpret it as:

`min(m, max(n1, n2, ..., n))`

or...

`min(m, sum(n1, n2, ..., n))`

Reply:
I'm inclined to not handled `SET_OP_INTERSECTION_PRIMARY_ALL` as part of this PR because I think figuring out which of those 2 behaviours is the "correct" one seems like it's own thing.

Reply:
Agree but to confirm: the only new additions will be `SET_OP_MINUS_PRIMARY_ALL` and `SET_OP_INTERSECTION_MULTISET_ALL`

Reply:
> Agree but to confirm: the only new additions will be SET_OP_MINUS_PRIMARY_ALL and SET_OP_INTERSECTION_MULTISET_ALL

Yes, we should leave the other variants for their own PR (if anyone ends up needing/wanting them).